src/main/resources/assets/admin/common/js/ai/AiHelper.ts

```diff
@@ -74,17 +79,26 @@ export class AiHelper {
     }
 
     private getOrCreateAiIcon(): AiDialogControl {
-        if (AiHelper.DIALOG_CONTROL_REGISTRY.has(this.config.aiButtonContainer)) {
-            return AiHelper.DIALOG_CONTROL_REGISTRY.get(this.config.aiButtonContainer);
+        if (AiHelper.getOrCreateIconsRegistry().has(this.config.aiButtonContainer)) {
+            return AiHelper.getOrCreateIconsRegistry().get(this.config.aiButtonContainer);
         }
 
         const aiIcon = new AiDialogControl();
         this.config.aiButtonContainer.appendChild(aiIcon);
-        AiHelper.DIALOG_CONTROL_REGISTRY.set(this.config.aiButtonContainer, aiIcon);
+        AiHelper.getOrCreateIconsRegistry().set(this.config.aiButtonContainer, aiIcon);
 
         return aiIcon;
     }
 
+    private static getOrCreateIconsRegistry(): Map<Element, AiDialogControl> {
+        if (!Store.instance().has(AI_ICONS_REGISTRY_KEY)) {
+            console.log('Creating new AI icons registry');
+            Store.instance().set(AI_ICONS_REGISTRY_KEY, new Map());
```

Comment:
We need to rewrite this and use `WeakMap` here, since keys are objects. The thing is, objects won't be release and garbage collected, even if there are no reference to them left.
The bad part: we won't be able to use `forEach` on WeakMap. This, instead of iteration, we better switch to event-based updates. But must be careful with the implementation: events could fire/handled in different order, so I suggest events to signal only state update, while the actual state will be stored here and accessed in AiDialogControl from single place (we can store it here, in AiHelper).